src/engine.rs

```diff
@@ -65,23 +65,25 @@ impl PromptWidget {
     }
 }
 
-trait EventParser {
+trait InputParser {
     fn parse_event(&self, event: Event) -> ReedlineEvent;
+    fn update_keybindings(&mut self, keybindings: Keybindings);
+    fn edit_mode(&self) -> EditMode;
 }
```

Comment:
This is a makeshift interface for now. It should remain the same in spirit only (i.e. it parses `Event`s to `ReedlineEvent`s) but the interface will definitely need updating.

Reply:
Note: Now the user of this lib can develop their own InputParser's without any help from the lib. All they need to do is adhere to this interface and we will be done. (prompt handling might pose a caveat in this though)